### ISSUE TITLE
Default value of 0 for integer results in a random value.

### ISSUE THREAD

**Issue by lnmp4000** [pact-foundation/pact-js#742]
This is an issue in Pact JS V3 `MatchersV3`
Attempting to pass `0` as the default value for the `integer` matcher results in a random value instead of the required value.

The `if(int)` test fails for the value `0` because this is equivalent to `false` in js.

```
/**
 * Value must be an integer (must be a number and have no decimal places)
 * @param int Example value. If omitted a random value will be generated.
 */
export const integer = (int?: number): Matcher<number> => {
  if (int) {
    return {
      'pact:matcher:type': 'integer',
      value: int,
    };
  }
  return {
    'pact:generator:type': 'RandomInt',
    'pact:matcher:type': 'integer',
    value: 101,
  };
};
```

The following results in a random number for `lowerBound` instead of zero

```
results: atLeastOneLike({
    id: integer(1),
    lowerBound: integer(0),
    upperBound: integer(10)
  })
```

**Comment by mefellows** [comment]
Ah, nice pickup  @lnmp4000. Thanks for the report, we can get this fixed in the next release.

**Comment by TimothyJones** [comment]
Thanks for the report! This will be fixed in v10.0.0-beta.50 once it has finished releasing.
